File: src/main/java/upbrella/be/user/dto/response/SingleUserInfoResponse.java
package upbrella.be.user.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.user.entity.User;

@Getter
@Builder
public class SingleUserInfoResponse {

    private long id;
    private String name;
    private String phoneNumber;
    private String email;
    private String bank;
    private String accountNumber;
    private boolean adminStatus;

    public static SingleUserInfoResponse fromUser(User user) {

        return SingleUserInfoResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .phoneNumber(user.getPhoneNumber())
                .email(user.getEmail())
                .bank(user.getBank())
                .accountNumber(user.getAccountNumber())
                .adminStatus(user.isAdminStatus())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/SingleUserInfoResponse.kt
package upbrella.be.user.dto.response

import upbrella.be.user.entity.User
import java.time.LocalDateTime

data class SingleUserInfoResponse(
    val id: Long,
    val name: String,
    val phoneNumber: String,
    val email: String,
    val bank: String?,
    val accountNumber: String?,
    val adminStatus: Boolean,
    val createdAt: LocalDateTime,
) {
    companion object {
        fun fromUser(user: User): SingleUserInfoResponse {
            return SingleUserInfoResponse(
                id = user.id!!,
                name = user.name,
                phoneNumber = user.phoneNumber,
                email = user.email,
                bank = user.bank,
                accountNumber = user.accountNumber,
                adminStatus = user.adminStatus,
                createdAt = user.createdAt!!
            )
        }
    }
}
